### ISSUE TITLE
config in folder level not substituted in commad

### ISSUE THREAD

**Issue by Penguinang** [augustocdias/vscode-shell-command#27]
key from settings.json in folder level not get substituted in command, but it works in workspace level's settings
VSC version: 1.60.1
extension version: 1.5.0

**Comment by augustocdias** [comment]
Can you provide more information please?

your tasks and input configuration would be a good start.

**Comment by Penguinang** [comment]
1/.vscode/tasks.json
```
{
    "version": "2.0.0",
    "inputs": [
        {
            "id": "test_va",
            "type": "command",
            "command": "shellCommand.execute",
            "args": {
                "command": "echo ${config:test_va}",
                "useSingleResult": true
            }
        }
    ],
    "tasks": [
        {
            "label": "My Task",
            "command": "echo ${input:test_va}",
            "type": "shell",
            "args": [],
        }
    ]
}
```
1/.vscode/settings.json
```
{
    "test_va": "test_va"
}
```
xxx.code-workspace
```
{
	"folders": [
		{
			"path": "1"
		},
		{
			"path": "2"
		}
	]
}
```
If I open the multiroot workspace and run task `My Task`, I got empty picker list panel.
If I open the folder itself, I can successfully echo the test_va

**Comment by Penguinang** [comment]
It seems that the `getConfiguration()` call doesn't consider the folder's .vscode/settings.json entries, though the API document says it's a merged view of all settings...Is that true? I'm new to vscode extension API...

**Comment by augustocdias** [comment]
From what I could understand from the documentation it should be in the root folder of your workspace. If inside any sub folder it won't work.

**Comment by augustocdias** [comment]
Can you post your project folder structure?

**Comment by Penguinang** [comment]
This is tree of my workspace, I have two folder 1 and 2. In folder 1, I have some task and coressponding settings in folder1's settings.json. Then for some reason I add the folder 2 to the workspace, and create a .code-workspace folder, the tasks in folder1's tasks.json which depend on settings through shellCommand extension in folder1's settings.json are not working now....
```
.
├── 1
│   ├── .vscode
│   │   ├── settings.json
│   │   └── tasks.json
│   └── a
├── 2
│   ├── .vscode
│   │   └── tasks.json
│   └── a
└── test.code-workspace

4 directories, 6 files
```
2/.vscode/tasks.json
```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "My Task",
            "command": "echo hello",
            "type": "shell",
            "args": [],
            "problemMatcher": [
                "$tsc"
            ],
            "presentation": {
                "reveal": "always"
            },
            "group": "build"
        }
    ]
}
```

**Comment by Penguinang** [comment]
I think maybe it's because the `getConfiguration()` call doesn't merge the folder's settings... If I merge the settings manually like below, the problem get solved...
```javascript
    bindWorkspaceConfigVariable(value) {
        let matchResult = this.configVarRegex.exec(value);
        if (!matchResult) {
            return '';
        }
        // Get value from workspace configuration "settings" dictionary
        let result = vscode.workspace.getConfiguration().get(matchResult[1], '');
        if (result) {
            return result;
        }

        result = vscode.workspace.getConfiguration(null, vscode.window.activeTextEditor.document).get(matchResult[1]);
        if (result) {
            return result;
        }

        for (let index = 0; index < vscode.workspace.workspaceFolders.length; index++) {
            const w = vscode.workspace.workspaceFolders[index];
            result = vscode.workspace.getConfiguration(null, w).get(matchResult[1]);
            if (result) {
                return result;
            }
        }
        return "";
    }
```

**Comment by augustocdias** [comment]
Do you want to submit a PR or should I do it?

**Comment by Penguinang** [comment]
Yeah, I will submita PR, thanks.

**Comment by Penguinang** [comment]
Fixed in #28

**Comment by augustocdias** [comment]
New version is already available. Thanks for the PR
